shapely/strtree.py

```diff
@@ -34,317 +34,185 @@ class STRtree:
+        input geometry index and tree geometry index, simply transpose the
+        results.
+
+        If a predicate is provided, the tree geometries are further filtered to
+        those that meet the predicate when comparing the input geometry to the
+        tree geometry:
+        predicate(geom, tree_geometry)
 
         The 'dwithin' predicate requires GEOS >= 3.10.
 
-        This returns an array with shape (2,n) where the subarrays correspond
-        to the indexes of the input geometries and indexes of the tree geometries
-        associated with each.  To generate an array of pairs of input geometry
-        index and tree geometry index, simply transpose the results.
-
-        In the context of a spatial join, input geometries are the "left" geometries
-        that determine the order of the results, and tree geometries are "right" geometries
-        that are joined against the left geometries.  This effectively performs
-        an inner join, where only those combinations of geometries that can be joined
-        based on envelope overlap or optional predicate are returned.
+        In the context of a spatial join, input geometries are the "left"
+        geometries that determine the order of the results, and tree geometries
+        are "right" geometries that are joined against the left geometries.
+        This effectively performs an inner join, where only those combinations
+        of geometries that can be joined based on overlapping extents or optional
+        predicate are returned.
 
         Any geometry that is None or empty in the input geometries is omitted from
         the output.
 
         Parameters
         ----------
         geometry : Geometry or array_like
-            Input geometries to query the tree.  The envelope of each geometry
-            is automatically calculated for querying the tree.
+            Input geometries to query the tree and filter results using the
+            optional predicate.
         predicate : {None, 'intersects', 'within', 'contains', 'overlaps', 'crosses',\
 'touches', 'covers', 'covered_by', 'contains_properly', 'dwithin'}, optional
             The predicate to use for testing geometries from the tree
-            that are within the input geometry's envelope.
+            that are within the input geometry's extent.
         distance : number or array_like, optional
             Distances around each input geometry within which to query the tree
             for the 'dwithin' predicate.  If array_like, shape must be
             broadcastable to shape of geometry.  Required if predicate='dwithin'.
 
         Returns
         -------
-        ndarray with shape (2, n)
-            The first subarray contains input geometry indexes.
-            The second subarray contains tree geometry indexes.
+        ndarray with shape (n,) if geometry is a scalar
+            Contains tree geometry indices.
+
+        OR
```

Comment:
(not fully sure if this will render nicely with sphinx, will do a check locally)

Reply:
With the new theme, this actually looks OK (it considers the "OR" also as a possible return value, which makes it bold, but visually that doesn't look problematic)